### PR TITLE
Add more missing math functions

### PR DESCRIPTION
**Pull request by kennethbgoodin**
As per issue #667 add missing functions for `atan2`, `cosh`, `sinh`, and `tan`. Also add f32 calls and tests for `cbrt` and `hypot`. Note that this does not make the `num` crates fully testable (ex: #770 )

This could be further cleaned up by condensing the f32 and f64 variants of each function into the same match arm. Or we could reduce even more code duplication wrt reading/writing floats similar to how intrinsics does so here: https://github.com/RalfJung/miri/blob/6a0d092eaf3150a4ae2e509744c49d5cadc60efc/src/intrinsic.rs#L188

### CODE REVIEW

**Comment by kennethbgoodin** [comment]
Awesome. I wasn't sure about the windows name for `hypotf` so we'll see.

**Comment by RalfJung** [comment]
Oh also, if you want to try something more like https://github.com/RalfJung/miri/blob/6a0d092eaf3150a4ae2e509744c49d5cadc60efc/src/intrinsic.rs#L188 that'd be great.

**Comment by RalfJung** [comment]
CI looks happy. @kennethbgoodin did you want to try to share the float input/output code some more?

**Comment by kennethbgoodin** [comment]
Yes I am.

File: src/fn_call.rs [hunk 1]
```diff
@@ -560,18 +560,56 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
...
+            // math functions
+
+            "cbrtf" | "coshf" | "sinhf" |"tanf" => {
+                // FIXME: Using host floats.
+                let f = f32::from_bits(this.read_scalar(args[0])?.to_u32()?);
+                let f = match link_name {
+                    "cbrtf" => f.cbrt(),
+                    "coshf" => f.cosh(),
+                    "sinhf" => f.sinh(),
+                        "tanf" => f.tan(),
```
Comment (RalfJung) [inline src/fn_call.rs]: Indentation looks wrong here.

**Comment by RalfJung** [comment]
Looks good except for the indentation.

In the future, please don't force-push unless you are resolving conflicts with a rebase.

**Pull request merged**